packages/ra-ui-materialui/src/input/SearchInput.tsx

```diff
@@ -8,17 +8,18 @@ import { CommonInputProps } from './CommonInputProps';
 import { TextInput, TextInputProps } from './TextInput';
 
 export const SearchInput = (props: SearchInputProps) => {
+    const { label, ...rest } = props;
```

Comment:
I don't understand why you made this change

Reply:
To handle the case when specifying `undefined` for a SearchInput's label explicitly (like `<SearchInput source="q" alwaysOn label={undefined} />`) will result in the default label is being displayed.
[Reproduction on codesandbox](https://codesandbox.io/s/react-admin-explicit-undefined-label-miiic7?file=/src/posts/PostList.tsx)

Reply:
Also noticed, that it's possible to use SearchInput with `0` label like:
```js
<SearchInput source="q" alwaysOn label={0} />
```

I think, it would be better to revert all changes in this PR, that related to the SearchInput's label overwriting logic and open a new issue to discuss, what and how to do about this case (IMHO it's better to prevent users from providing a `label` at all instead of throwing an error).